### PR TITLE
Update Travis config for Xcode 8.3 and fix test failures

### PR DESCRIPTION
**Pull request by liuxuan30**
update device id in Travis CI and xcode 8.3 image

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -39,8 +39,8 @@ def devices
   {
     iOS: {
       sdk: 'iphonesimulator',
-      device: "id='22FA2149-1241-469C-BF6D-462D3837DB72'",
-      uuid: '22FA2149-1241-469C-BF6D-462D3837DB72'
+      device: "34FA4749-C467-4D45-9F8E-E31AEDDC39C2'",
```
Comment (petester42) [inline Rakefile]: The `id='` is needed here.

Comment (liuxuan30) [inline Rakefile]: done.

**Comment by liuxuan30** [comment]
@petester42 I guess we need to use Xcode 8.3 image as well? What else to we have to modify? I can do it all in one. Just found what you did for Xcode 8.2: https://github.com/danielgindi/Charts/commit/99e9aa04324739b531fe0bdbe7265442e3545344

But I'm not sure if I just change `osx_image: xcode8.2` to `xcode8.3` is enough? I saw you update `xcodeproj` as well.

**Comment by petester42** [comment]
ya. just change the config to be 8.3

**Comment by liuxuan30** [comment]
OK, I have changed to 8.3. Let's see if it can pass

**Comment by liuxuan30** [comment]
```
▸ Processing Info.plist

▸ Touching ChartsTests.xctest

▸ Signing /Users/travis/Library/Developer/Xcode/DerivedData/Charts-dsitxzqjkujfnvaoreewgudaitlo/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/ChartsTests.xctest

2017-04-20 02:46:56.978 xcodebuild[4087:13500] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}

Testing failed:

	Test target ChartsTests encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)

** TEST FAILED **
```

I tried on my machine with `rake ci[iOS]`, it has same error code, and I also checked the log:
```
106 11:00:46.292 xcodebuild[82088:12196011] Tracking pid 82380 for simulator <DVTiPhoneSimulator: 0x7fb8ca6c5d80> {
107         SimDevice: SimDevice : iPhone 7 (1A98B8F5-C2C4-451B-B4C0-4FC506ACC3D6) : state={ Booted } deviceType={ SimDeviceType : com.apple.        CoreSimulator.SimDeviceType.iPhone-7 } runtime={ SimRuntime : 10.3 (14E269) - com.apple.CoreSimulator.SimRuntime.iOS-10-3 }
108 }
109 11:00:46.292 xcodebuild[82088:12196011] Checking to see if pid 82380 is valid: kill(pid, 0) = 0 (YES, it's still running)
110 11:00:46.292 xcodebuild[82088:12196011] Setting up DVTTaskNotifyOnProcDeath for pid 82380
111 11:00:46.292 xcodebuild[82088:12190492] Test process runnable PID is 82380.
112 11:00:46.507 xcodebuild[82088:12190492] Got death notice for pid 82380, removing from SimulatorSessionMap
113 11:00:46.507 xcodebuild[82088:12196014] 1A98B8F5-C2C4-451B-B4C0-4FC506ACC3D6: Executable terminated at path: /Applications/Xcode.app/            Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Xcode/Agents/xctest (status = 20992)
114 11:00:46.508 xcodebuild[82088:12190492] Test operation failure: Launch session expired before checking in.
115 11:00:46.509 xcodebuild[82088:12190492] _finishWithError:Error Domain=IDETestOperationsObserverErrorDomain Code=5 "Launch session expired        before checking in." UserInfo={NSLocalizedDescription=Launch session expired before checking in.} didCancel: 1   
```

Any idea? @petester42  I googled but can't find the exact error code solution

**Comment by petester42** [comment]
I don't have time right now but I'll look at it maybe tomorrow

**Comment by liuxuan30** [comment]
Sure. I also tried to manually run 
```
xcodebuild -project 'Charts.xcodeproj' -scheme 'ChartsTests' -configuration 'Debug' -sdk iphonesimulator -destination id='E40B5365-EF82-430D-A767-2A37995CCEE1' clean build test
```
on my machine, it reports:
```
  /bin/sh -c /Users/xuanliu/Library/Developer/Xcode/DerivedData/Charts-ambsdrskvocwekcmoobyhanaiixm/Build/Intermediates/CodeCoverage/Intermediates/Charts.build/Debug-iphonesimulator/ChartsTests.build/Script-064DBAE11D93813D002584CD.sh
Depencies have already been built.

2017-04-20 11:53:08.710 xctest[87783:12569635] The bundle “ChartsTests” couldn’t be loaded because it is damaged or missing necessary resources. Try reinstalling the bundle.
2017-04-20 11:53:08.711 xctest[87783:12569635] (dlopen_preflight(/Users/xuanliu/Library/Developer/Xcode/DerivedData/Charts-ambsdrskvocwekcmoobyhanaiixm/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/ChartsTests.xctest/ChartsTests): Library not loaded: @rpath/FBSnapshotTestCase.framework/FBSnapshotTestCase
  Referenced from: /Users/xuanliu/Library/Developer/Xcode/DerivedData/Charts-ambsdrskvocwekcmoobyhanaiixm/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator/ChartsTests.xctest/ChartsTests
  Reason: image not found)
2017-04-20 11:53:08.725 xcodebuild[87517:12565751] Error Domain=IDETestOperationsObserverErrorDomain Code=6 "Early unexpected exit, operation never finished bootstrapping - no restart will be attempted" UserInfo={NSLocalizedDescription=Early unexpected exit, operation never finished bootstrapping - no restart will be attempted}
2017-04-20 11:53:08.726 xcodebuild[87517:12567557] Connection peer refused channel request for "dtxproxy:XCTestManager_IDEInterface:XCTestManager_DaemonConnectionInterface"; channel canceled <DTXChannel: 0x7fba8660af50>

Testing failed:
	Test target ChartsTests encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted)
** TEST FAILED **

```
Not sure if it's related. I tried to link the framework in ChartTests build phases, but no luck.

**Comment by liuxuan30** [comment]
Today, I just tried to manually copy `FBSnapshotTestCase.framework` into 
>/Users/xuanliu/Library/Developer/Xcode/DerivedData/Charts-ambsdrskvocwekcmoobyhanaiixm/Build/Intermediates/CodeCoverage/Products/Debug-iphonesimulator

```
xuanliu:Debug-iphonesimulator xuanliu$ ls
Charts.framework		ChartsTests.swiftmodule		ChartsTests.xctest		FBSnapshotTestCase.framework
```
and just call
>xcodebuild -project 'Charts.xcodeproj' -scheme 'ChartsTests' -configuration 'Debug' -sdk iphonesimulator -destination id='E40B5365-EF82-430D-A767-2A37995CCEE1' test

It actually succeeded. Still investigating why it did not copy the framework

**Comment by liuxuan30** [comment]
Finally all green. @petester42 if the new commits of this PR looks good to you, I will merge it.

File: scripts/copy-carthage-frameworks.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    framework=$(basename "${!VAR}")
+    export SCRIPT_INPUT_FILE_$n="$SRCROOT"/Carthage/Build/$plat/"$framework".framework
+done
+
+/usr/local/bin/carthage copy-frameworks || exit
+
+for (( n = 0; n < SCRIPT_INPUT_FILE_COUNT; n++ )); do
+    VAR=SCRIPT_INPUT_FILE_$n
+    source=${!VAR}.dSYM
+    dest=${BUILT_PRODUCTS_DIR}/$(basename "$source")
```
Comment (liuxuan30) [inline scripts/copy-carthage-frameworks.sh]: BTW, which dSYM is here? and what's for? carthage copy-framework seems will copy `FBSnapshotTestCase.framework.dSYM` under `CodeCoverage/Products/Debug-iphonesimulator/`

**Comment by liuxuan30** [comment]
@petester42 sorry to bring this up again, don't know why but I notice travis will compile Charts twice, but on my machine, it seems just once.

Do you think we can change `clean build test` to just `clean test`?

**Pull request merged**